tests/macro_expansion/test_macro_expansion.py

```diff
@@ -27,35 +27,21 @@ def setUp(self):
 
     def test_macro_expansion(self):
         """macro_expansion/macro_expansion.yaml"""
-        files = [
-            "defined_undefined_test.cpp",
-            "infinite_loop_test.cpp",
-            "function_like_test.cpp",
-            "max_level.cpp",
-        ]
-        codebase = {
-            "files": [
-                os.path.realpath(os.path.join(self.rootdir, f)) for f in files
-            ],
-            "platforms": ["CPU"],
-            "exclude_files": set(),
-            "exclude_patterns": [],
-            "rootdir": self.rootdir,
-        }
+        codebase = CodeBase(self.rootdir)
         cpu_entries = []
         gpu_entries = []
-        for f in files:
+        for f in list(codebase):
```

Comment:
Interesting that this works - it just converts `__iter__` into a list under the hood?

Reply:
Yes, exactly.  I think it's the last constructor in the list [here](https://docs.python.org/3/library/stdtypes.html#list); defining `__iter__` makes a `CodeBase` iterable, so we can convert it to a `list`.